src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -849,8 +849,21 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                     AST::keyword_t* kwargs=nullptr, size_t n_kwargs=0) {
         if (intrinsic_node_handler.is_present(call_name)) {
             return intrinsic_node_handler.get_intrinsic_node(call_name, al, loc,
-                    args, ann_assign_target_type);
+                    args);
         }
+
+        if (call_name == "list" && (args.size() == 0 ||  args[0].m_value == nullptr)) {
+            if (ann_assign_target_type) {
+                ASR::ttype_t *type = ASRUtils::get_contained_type(ann_assign_target_type);
+                ASR::ttype_t* list_type = ASRUtils::TYPE(ASR::make_List_t(al, loc, type));
+                Vec<ASR::expr_t*> list;
+                list.reserve(al, 1);
+                return ASR::make_ListConstant_t(al, loc, list.p,
+                    list.size(), list_type);
```

Comment:
Is this creating an empty list constant?

Reply:
Yes